event_registration_cancel_reason/wizard/event_registration_cancel_log_reason.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+# © 2016 Antiun Ingeniería S.L.
+# © 2016 Pedro M. Baeza <pedro.baeza@serviciosbaeza.com>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import api, fields, models
+
+
+class EventRegistrationCancelLogReason(models.TransientModel):
+    _name = 'event.registration.cancel.log.reason'
+
+    event_type_id = fields.Many2one(
+        comodel_name="event.type", string="Event type")
+    reason_id = fields.Many2one(
+        comodel_name="event.registration.cancel.reason", required=True,
+        domain="['|', "
+               " ('event_type_ids', '=', False), "
+               " ('event_type_ids', '=', event_type_id)]")
+
+    @api.model
+    def default_get(self, var_fields):
+        res = super(EventRegistrationCancelLogReason, self).default_get(
+            var_fields)
+        registration = self.env['event.registration'].browse(
+            self.env.context['active_id'])
+        res['event_type_id'] = registration.event_id.type.id
+        return res
+
+    @api.multi
+    def button_log(self):
+        self.ensure_one()
```

Comment:
@pedrobaeza I know.
this is why I suggest the roadmap to add the feature accessible in tree view. For me it makes sense to be able possibility to mass cancel registration selecting registrations in the tree view (with multi-select/Action/Cancel Registration) with similar effect as clicking on the red cross in the following view and adding a reason during the wizard.
![Uploading Screenshot from 2016-02-14 21-12-03.png…]()